document-store/src/main/java/org/hypertrace/core/documentstore/expression/impl/ArrayIdentifierExpression.java

```diff
@@ -14,28 +14,132 @@
+  // Package-private: used internally by factory methods
+  static ArrayIdentifierExpression of(String name, FlatCollectionDataType arrayElementType) {
+    return new ArrayIdentifierExpression(name, arrayElementType);
   }
 
-  /** Returns the array type if specified, empty otherwise */
-  public Optional<ArrayType> getArrayType() {
-    return Optional.ofNullable(arrayType);
+  public static ArrayIdentifierExpression ofStrings(final String name) {
+    return of(name, PostgresDataType.TEXT);
+  }
+
+  public static ArrayIdentifierExpression ofInts(final String name) {
+    return of(name, PostgresDataType.INTEGER);
+  }
+
+  public static ArrayIdentifierExpression ofLongs(final String name) {
+    return of(name, PostgresDataType.BIGINT);
+  }
+
+  public static ArrayIdentifierExpression ofShorts(final String name) {
+    return of(name, PostgresDataType.SMALLINT);
+  }
+
+  public static ArrayIdentifierExpression ofFloats(final String name) {
+    return of(name, PostgresDataType.FLOAT);
+  }
+
+  public static ArrayIdentifierExpression ofDoubles(final String name) {
+    return of(name, PostgresDataType.DOUBLE);
+  }
+
+  public static ArrayIdentifierExpression ofDecimals(final String name) {
+    return of(name, PostgresDataType.NUMERIC);
+  }
+
+  public static ArrayIdentifierExpression ofBooleans(final String name) {
+    return of(name, PostgresDataType.BOOLEAN);
+  }
+
+  public static ArrayIdentifierExpression ofTimestamps(final String name) {
+    return of(name, PostgresDataType.TIMESTAMP);
+  }
+
+  public static ArrayIdentifierExpression ofTimestampsTz(final String name) {
+    return of(name, PostgresDataType.TIMESTAMPTZ);
+  }
+
+  public static ArrayIdentifierExpression ofDates(final String name) {
+    return of(name, PostgresDataType.DATE);
+  }
+
+  public static ArrayIdentifierExpression ofUuids(final String name) {
+    return of(name, PostgresDataType.UUID);
+  }
+
+  public static ArrayIdentifierExpression ofBytesArray(final String name) {
+    return of(name, PostgresDataType.BYTEA);
+  }
+
+  // Package-private: used internally by getPostgresArrayTypeString()
+  Optional<FlatCollectionDataType> getArrayElementType() {
+    return Optional.ofNullable(arrayElementType);
+  }
+
+  /**
+   * Returns the PostgreSQL array type string for this expression, if the element type is specified.
+   *
+   * <p>Examples:
+   *
+   * <ul>
+   *   <li>TEXT → "text[]"
+   *   <li>INTEGER → "integer[]"
+   *   <li>BOOLEAN → "boolean[]"
+   *   <li>FLOAT → "real[]"
+   *   <li>DOUBLE → "double precision[]"
+   * </ul>
+   *
+   * @return Optional containing the PostgreSQL array type string, or empty if no type is specified
+   * @throws IllegalArgumentException if the element type is not a PostgresDataType
+   */
+  public Optional<String> getPostgresArrayTypeString() {
+    return getArrayElementType().map(this::toPostgresArrayType);
+  }
+
+  private String toPostgresArrayType(FlatCollectionDataType type) {
```

Comment:
Having the name "toPostgres..." in a generic class breaks abstraction though it is private. If it's possible, I'd avoid it.